docs/installation/server-config/security.rst

```diff
@@ -0,0 +1,163 @@
+.. _ksql-security:
+
+Securing KSQL
+-------------
+
+To connect to a secured Kafka cluster, Kafka client applications must provide their security credentials. In the same way,
+KSQL is configured so that the KSQL servers are authenticated and authorized, and data communication is encrypted when
+communicating with the Kafka cluster. KSQL can be configured for:
+
+- :ref:`SSL for encryption <kafka_ssl_encryption>`
+- :ref:`SASL for authentication <kafka_sasl_auth>`
+- :ref:`HTTPS for Confluent Schema Registry <schema_registry_http_https>`
+
+-------------------------------------------------
+Using KSQL with an Apache Kafka Secured with ACLs
+-------------------------------------------------
+
+You can use KSQL with Apache Kafka clusters that are secured with ACLs. The behavior depends on whether the cluster is
+interactive or non-interactive.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Interactive KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Interactive KSQL clusters require that the KSQL user has open access to create, read, write and delete topics and
+use any consumer group.
+
+The required ACLs are:
+
+- *DESCRIBE_CONFIGS* permission on the *CLUSTER*.
+- *CREATE* permission on the *CLUSTER*.
+- DESCRIBE*, *READ*, *WRITE* and *DELETE* permissions on the *<any>* *TOPIC*.
+- *DESCRIBE* and *READ* permissions  on the *<any>* *GROUP*.
+
+It is still possible to restrict the KSQL user from accessing specific resources using *DENY* ACLs. For example, you can add a
+*DENY* ACL to stop KSQL queries from accessing a topic that contains sensitive data.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Non-Interactive KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Non-interactive KSQL clusters will run with much more restrictive ACLs, though it currently requires a little effort to
+work out what ACLs are required. This will be improved in future KSQL releases.
+
+Standard ACLs
+    The KSQL user always requires:
+
+    - *DESCRIBE_CONFIGS* permission on the *CLUSTER*.
+    - *DESCRIBE* permission on the *__consumer_offsets* topic.
+
+    If you want KSQL to create internal and sink topics then the KSQL user should also be granted:
+
+    - *CREATE* permission on the *CLUSTER*.
+
+Source topics
+    KSQL users require *DESCRIBE* and *READ* permissions for each source and input topic. The topic should already exist
```

Comment:
Source vs. input, and sink vs. output.

Could we stick to one terminology here please?  I'd go with "input topic" and "output topic" (rather than source and sink), because, for example, we are almost always using this wording elsewhere, unless we are e.g. discuss internals of the Processor API, for example.

Reply:
@miguno if i replace `source` for `input`,  in this example the sentence would become incorrect, e.g.: `KSQL users require *DESCRIBE* and *READ* permissions for each input and input topic.`

Reply:
That's exactly why I asked -- my understanding (might be wrong) is that source==input.  So the sentence should be (I think) simply: `KSQL users require *DESCRIBE* and *READ* permissions for each input topic`.

I'd be surprised if there's a difference between source topic and input topic.  cc @big-andy-coates